nbsphinx_link/__init__.py

```diff
@@ -176,28 +174,28 @@ class LinkedNotebookParser(NotebookParser):
 
     supported = 'linked_jupyter_notebook',
 
-    def parse(self, inputstring, document):
+    def parse(self, inputstring: str, document: ddocument) -> None:
         """Parse the nblink file.
 
         Adds the linked file as a dependency, read the file, and
         pass the content to the nbshpinx.NotebookParser.
         """
         link = json.loads(inputstring)
         env = document.settings.env
-        source_dir = os.path.dirname(env.doc2path(env.docname))
+        source_file = Path(env.docname)
+        source_dir = source_file.parent
 
-        abs_path = os.path.normpath(os.path.join(source_dir, link['path']))
-        path = utils.relative_path(None, abs_path)
+        abs_path = (env.srcdir / source_dir / link['path']).resolve()
+        path = Path(utils.relative_path(None, abs_path))
 
         extra_media = link.get('extra-media', None)
         if extra_media:
-            source_file = env.doc2path(env.docname)
             collect_extra_media(extra_media, source_file, path, document)
 
         register_dependency(path, document)
 
-        target_root = env.config.nbsphinx_link_target_root
-        target = utils.relative_path(target_root, abs_path).replace(os.path.sep, '/')
+        target_root = env.config.nbsphinx_link_target_root or Path.cwd()
+        target = abs_path.relative_to(target_root, walk_up=True)
```

Comment:
Above you wrap a `util.relative_path` call in `Path`, here you replace with pathlib method. Is this distinction significant? Do they behave differently?